src/theory/quantifiers/sygus/sygus_repair_const.cpp

```diff
@@ -228,9 +212,12 @@ bool SygusRepairConst::repairSolution(Node sygusBody,
   }
 
   Trace("sygus-engine") << "Repairing previous solution..." << std::endl;
-  // make the satisfiability query
+  // Make the satisfiability query. We use the "ALL" logic, since the subcall
+  // may e.g. introduce non-linear arithmetic in linear logics.
   std::unique_ptr<SolverEngine> repcChecker;
-  SubsolverSetupInfo ssi(d_env);
+  LogicInfo lall("ALL");
+  SubsolverSetupInfo ssi(
+      d_env.getOptions(), lall, d_env.getSepLocType(), d_env.getSepDataType());
```

Comment:
I don't understand this part. All the code that deals with logics is removed in this PR. So, how do we ensure that the repaired constant is within the current logic?

Reply:
The issue was never the repaired constant, it was the query.

For example, if we are looking to repair `(* c x) + d > 0` where `c` and `d` are generated via `(Constant Int)`, and the logic did not permit non-linear arithmetic, then `c` would not be allowed to be repaired.  We now just use `ALL` as the logic for this query, since `(* c x)` is non-linear. 

The repaired constant themselves are simply integer values.